### PR TITLE
ui: Refreshes parent route to fix secret state discrepancies within a namespace for kv v2 

### PR DESCRIPTION
**Pull request by hellobontempo**
#### bug description 🐛 
Deleting a secret **in a namespace** initially renders an empty state but if you navigate between sibling routes (click another tab) the stale secret model (`kv/data`) re-renders when you return to the details view (`Secret` tab). The deleted state restores when you refresh the page.
delete-bug

<hr>

### after fix 🔧 
delete-bug

### CODE REVIEW

File: ui/lib/kv/addon/components/page/secret/details.hbs
Comment (hellobontempo) [inline ui/lib/kv/addon/components/page/secret/details.hbs:1]: small cleanup task to remove the extra toggle action from the component file

File: ui/lib/kv/addon/components/page/secret/details.js [hunk 1]
```diff
@@ -106,6 +97,13 @@ export default class KvSecretDetails extends Component {
     }
   }
 
+  refreshRoute() {
```
Comment (hellobontempo) [inline ui/lib/kv/addon/components/page/secret/details.js:100]: consolidated to one method

File: ui/lib/kv/addon/routes/secret.js
Comment (hellobontempo) [inline ui/lib/kv/addon/routes/secret.js:1]: I opted to use the `willTransition` model hook because it is called when transitions happen between sibling routes. It is _not_ called when this route is initially rendered. I'm open to other ways of approaching. 

Another option is passing down an action through the controller. I tried going that way at first but I found previous implementations already hard to follow. It seemed pretty disjointed to pass an action from a controller, to a route template and then down to the component to call it.

File: ui/lib/kv/addon/routes/secret.js [hunk 2]
```diff
@@ -32,4 +33,15 @@ export default class KvSecretRoute extends Route {
...
     });
   }
+
+  @action
+  willTransition(transition) {
+    // refresh the route if transitioning to secret.index (which happens after delete, undelete or destroy)
+    // or transitioning from editing either metadata or secret data (creating a new version)
+    const performedToolbarAction = transition.to.name === 'vault.cluster.secrets.backend.kv.secret.index';
+    const afterEditing = transition.from.localName === 'edit';
+    if (performedToolbarAction || afterEditing) {
```
Comment (hellobontempo) [inline ui/lib/kv/addon/routes/secret.js:43]: I previously had this as the following conditional - which might be easier to read? wanted it to be clear what was happening here
```js
    const { to, from } = transition;
    if (to.name === 'vault.cluster.secrets.backend.kv.secret.index' || from.localName === 'edit') 
```

File: ui/lib/kv/addon/routes/secret.js [hunk 3]
```diff
@@ -32,4 +33,15 @@ export default class KvSecretRoute extends Route {
...
+      this.refresh();
+    }
+  }
```
Comment (hellobontempo) [inline ui/lib/kv/addon/routes/secret.js:46]: I previously had the conditional above as the following - which might be easier to read? wanted it to be clear what was happening here
```js
const { to, from } = transition;
if (to.name === 'vault.cluster.secrets.backend.kv.secret.index' || from.localName === 'edit') 
```

File: ui/lib/kv/addon/routes/secret.js [hunk 4]
```diff
@@ -32,4 +33,15 @@ export default class KvSecretRoute extends Route {
       metadata: this.fetchSecretMetadata(backend, path),
     });
   }
+
```
Comment (Monkeychip) [inline ui/lib/kv/addon/routes/secret.js:36]: nice naming and comments here.

**Review by Monkeychip — APPROVED** [review]
LGTM, but you and I rubber ducked a little on this, so I understand if you want another set of eyes.

**Review by hashishaw — APPROVED** [review]
Great work tracking this down!
